src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,527 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    value = 20 - ABS(value);
+
+    if (value > 10) {
+        value = 10;
+    }
+
+    this->headHitbox.dim.radius = sKarebabaHitbox2.dim.radius + (value * 2);
+    this->actor.shape.rot.x = 0xC000 - (value * 0x100);
+    this->actor.shape.rot.y += value * 0x40 * 11;
+    this->actor.posRot.pos.y = (Math_Sins(this->actor.shape.rot.x) * -60.0f) + this->actor.initPosRot.pos.y;
+
+    thing = Math_Coss(this->actor.shape.rot.x) * 60.0f;
+
+    this->actor.posRot.pos.x = (Math_Sins(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.x;
+    this->actor.posRot.pos.z = (Math_Coss(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.z;
+
+    if ((this->bodyHitbox.base.acFlags & 2) != 0) {
+        EnKarebaba_SetupDying(this);
+        func_80032C7C(globalCtx, &this->actor);
+        return;
+    }
+
+    if (this->actor.params == 0) {
+        EnKarebaba_SetupUpright(this);
+    }
+}
+
+void EnKarebaba_Dying(EnKarebaba* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f position;
+    Vec3f rotation;
+
+    Math_ApproxF(&this->actor.speedXZ, 0.0f, 0.1f);
+
+    if (this->actor.params == 0) {
+        Math_ApproxUpdateScaledS(&this->actor.shape.rot.x, 0x4800, 0x71C);
+        EffectSsHahen_SpawnBurst(globalCtx, &this->actor.posRot.pos, 3.0f, 0, 12, 5, 1, -1, 10, 0);
+        if (0.005f < this->actor.scale.x && ((this->actor.bgCheckFlags & 2) != 0 || (this->actor.bgCheckFlags & 8) != 0)) {
+            this->actor.scale.z = 0.0f;
+            this->actor.scale.y = 0.0f;
+            this->actor.scale.x = 0.0f;
+            this->actor.speedXZ = 0.0f;
+            this->actor.flags &= ~5;
+            EffectSsHahen_SpawnBurst(globalCtx, &this->actor.posRot.pos, 3.0f, 0, 12, 5, 15, -1, 10, 0);
+        }
+        if ((this->actor.bgCheckFlags & 2) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+            this->actor.params = 1;
+        }
+    } else if (this->actor.params == 1) {
+        Math_Vec3f_Copy(&position, &this->actor.posRot.pos);
+        rotation.z = Math_Sins(this->actor.shape.rot.x) * 20.0f;
+        rotation.x = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Sins(this->actor.shape.rot.y);
+        rotation.y = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Coss(this->actor.shape.rot.y);
+
+        for (i = 0; i < 4; i++) {
+            func_800286CC(globalCtx, &position, &sVecZero1, &sVecZero1, 500, 50);
+            position.x += rotation.x;
+            position.y += rotation.z;
+            position.z += rotation.y;
+        };
+
+        func_800286CC(globalCtx, &this->actor.initPosRot.pos, &sVecZero1, &sVecZero1, 500, 100);
+        EnKarebaba_SetupDeadItemDrop(this, globalCtx);
+    }
+}
+
+void EnKarebaba_DeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx) != 0 || this->actor.params == 0) {
+        EnKarebaba_SetupWait(this);
```

Comment:
Maybe Wait should be more explicit about "being alive again"? It's also needed to improve on the name for it to not be confused with Idle